src/app/dts/template-editor/template-editor.component.ts

```diff
@@ -19,22 +25,22 @@ export class TemplateEditorComponent extends UnsubscribeOnDestroyAdapter impleme
    */
   existingTemplate = false;
 
+  /**
+   * User setting to show/hide drag and drop data fields
+   */
+  dataFieldsVisible = localStorage.getItem('dataFieldsVisible') === null;
+
   /**
    * Used to display status messages
    */
   statusMessage = '';
 
   templateEditor: FormGroup;
 
-  DATAFIELDS = [{
-      name: '{{ student.fistName }}',
-      avatar: 'student'
-    },
-    {
-      name: '{{ course.name }}',
-      avatar: 'course'
-    }
-  ];
+  DATAFIELDSSTUDENT = (dataFieldsStudent as any).default;
+  DATAFIELDSCOURSE = (dataFieldsCourse as any).default;
+  DATAFIELDSSTANDARD = (dataFieldsStandardAcc as any).default;
```

Comment:
I'd use the new types for those fields:
```DATAFIELDSSTUDENT: DataField = (dataFieldsStudent as any).default;```
and perhaps would also create a type for `dataFieldsStudent ` and all so you don't have to do the cast ```(dataFieldsCourse as any)```

Reply:
I would recommend against setting the type in this case.  The type is "trivially inferred" when setting the value.  It isn't bad practice but it's redundant because it's inferred when assigned.

Reply:
I wouldn't call ```(dataFieldsStudent as any).default``` a trivial type inference. Using ```any``` to circumvent type checking is definitely not recommended, nor trivial.

Reply:
I'm a bit stuck trying to understand your suggestion and need some more clarification.  Will you please enlighten me as to what the "dataFieldsStudent" interface would look like? Thanks.

Reply:
I think that changing your imports syntax would remove the need for `any` and `default` as suggested here: https://stackoverflow.com/questions/49996456/importing-json-file-in-typescript. 
Then the extra interface wouldn't be needed. 
I'd still specify the data type as ```DATAFIELDSSTUDENT: DataField``` for the sake of clarity.
While ```isActive = true``` is self-explanatory in terms of data type,  ```DATAFIELDSSTUDENT = dataFieldsStudent``` isn't.

Reply:
Followed the suggesting noted above, created a type script definition file, and the suggesting works when accessing the file content in the typescript, however am unable to access the contents from the template.  Scheduled a meeting with you so I can show you what I mean.  Thought this approach of talking might be better.

Reply:
Ok, that works.

Reply:
Reworked structure of drag and drop fields to store in a TypeScript object instead of a JSON object.